pennylane/transforms/__init__.py

```diff
@@ -192,17 +158,60 @@
     ~transforms.poly_extrapolate
     ~transforms.richardson_extrapolate
 
-Transforms core
----------------
+Other transforms
+~~~~~~~~~~~~~~~~
```

Comment:
Can we:
- Make "transforms for error mitigation" be a `~` level section (currently `-` level)
- Move "decorators and utility functions" to below the "Transforms library" section (i.e., below "Transforms that act on quantum functions" but above the "transforms developer functions"

I know this is straying into the territory of restructuring the docs, but the above seems like some low-hanging fruit.

Reply:
Done, thanks!